src/Memoization.cpp

```diff
@@ -366,7 +366,7 @@ class InjectMemoization : public IRMutator {
 
     using IRMutator::visit;
 
-    void visit(const ProducerConsumer *op) {
+    void visit(const Realize *op) {
```

Comment:
All Producer and Consumer nodes for a given name are contained in a Realize for that name. Is this correct? I think this may be the invariant we want documented in IR.h.